zio-schema-avro/shared/src/test/scala/zio/schema/codec/AvroSchemaCodecTest.scala

```diff
@@ -0,0 +1,2020 @@
+
+  def isPrimitive[A](assertion: Assertion[Primitive[A]]): Assertion[Schema[_]] =
+    Assertion.isCase[Schema[_], Primitive[A]]("Primitive", {
+      case p: Primitive[_] => Try { p.asInstanceOf[Primitive[A]] }.toOption
+      case _               => None
+    }, assertion)
+
+  def isStandardType[A](standardType: StandardType[A]): Assertion[Schema[_]] =
+    isPrimitive[A](hasField("standardType", _.standardType, equalTo(standardType)))
+
+  def isPrimitiveType[A](assertion: Assertion[StandardType[A]]): Assertion[Schema[_]] =
+    isPrimitive[A](hasField("standardType", _.standardType, assertion))
+}
+
+object SpecTestData {
+
+  @AvroAnnotations.name("MyEnum")
+  sealed trait CaseObjectsOnlyAdt
+
+  object CaseObjectsOnlyAdt {
+    case object A extends CaseObjectsOnlyAdt
+    case object B extends CaseObjectsOnlyAdt
+
+    @AvroAnnotations.name("MyC")
+    case object C extends CaseObjectsOnlyAdt
+  }
+
+  @AvroAnnotations.avroEnum
+  sealed trait CaseObjectAndCaseClassAdt
+
+  object CaseObjectAndCaseClassAdt {
+    case object A extends CaseObjectAndCaseClassAdt
+    case object B extends CaseObjectAndCaseClassAdt
+
+    @AvroAnnotations.name("MyC")
+    case object C           extends CaseObjectAndCaseClassAdt
+    case class D(s: String) extends CaseObjectAndCaseClassAdt
+  }
+
+  sealed trait UnionWithNesting
+
+  object UnionWithNesting {
+    sealed trait Nested extends UnionWithNesting
+
+    object Nested {
+      case object A extends Nested
+      case object B extends Nested
+    }
+
+    @AvroAnnotations.name("MyC")
+    case object C           extends UnionWithNesting
+    case class D(s: String) extends UnionWithNesting
+  }
+
+  case class Record(s: String, b: Boolean)
+
+  @AvroAnnotations.name("MyNamedRecord")
+  case class NamedRecord(s: String, b: Boolean)
+
+  @AvroAnnotations.name("MyNamedFieldRecord")
+  case class NamedFieldRecord(@AvroAnnotations.name("myNamedField") s: String, b: Boolean)
+
+  @AvroAnnotations.name("NestedRecord")
+  case class NestedRecord(s: String, nested: NamedRecord)
+
+  @AvroAnnotations.name("Simple")
+  case class SimpleRecord(s: String)
+
+  object FullAdtTest {
+    sealed trait TopLevelUnion
+
+    object TopLevelUnion {
+      case class RecordWithPrimitives(
+        string: String,
+        bool: Boolean,
+        int: Int,
+        double: Double,
+        float: Float,
+        short: Short,
+        bigInt: BigInt,
+        bigDecimal: BigDecimal,
+        unit: Unit,
+        char: Char,
+        uuid: UUID
+      ) extends TopLevelUnion
+      case class NestedRecord(innerRecord: InnerRecord)               extends TopLevelUnion
+      case class Unions(union: Union)                                 extends TopLevelUnion
+      case class Enumeration(enum: Enum)                              extends TopLevelUnion
+      case class Iterables(list: List[String], map: Map[String, Int]) extends TopLevelUnion
+
+      // TODO: Schema derivation fails for the following case classes
```

Comment:
is there an open issue for this already (bug or feature)?

Reply:
The java time schemas take a formatter so we only provide default implicit for these in the `zio.schema.DefaultJavaTimeSchemas` trait. So you need to mix that trait in for the implicit derivation to work here I believe.